activerecord/lib/active_record/connection_adapters/connection_specification.rb

```diff
@@ -17,6 +17,26 @@ def to_hash
         @config.merge(name: @name)
       end
 
+      # Converts a one level connection config, to the new format
+      # which is a two level, such as: {'primary' => {'adapter' => 'foo'}}
+      class LegacyConfigTransformer # :nodoc:
```

Comment:
I don't think we should call it a legacy config. It's still a valid config that most of Rails apps are going to use, as long as we don't deprecate one level config.